includes/helper/class-wp-job-manager-helper.php

```diff
@@ -194,65 +194,158 @@ public function addon_core_version_check( $do_check, $minimum_required_core_vers
 	/**
 	 * Check for licence managed WPJM addon plugin updates.
 	 *
-	 * @param array $check_for_updates_data
+	 * @param object $check_for_updates_data
 	 *
-	 * @return array
+	 * @return object
 	 */
 	public function check_for_updates( $check_for_updates_data ) {
-		$available_addon_updates = [];
+		$installed_plugins = $this->get_installed_plugins( false, true );
+		$updates           = $this->get_plugin_update_info( $installed_plugins );
+
+		$notice_data = [];
+
 		// Set version variables.
-		foreach ( $this->get_installed_plugins() as $product_slug => $plugin_data ) {
-			$response = $this->get_plugin_version( $plugin_data['_filename'] );
+		foreach ( $installed_plugins as $filename => $plugin_data ) {
+			$wpjmcom_plugin_slug = $plugin_data['_product_slug'];
+			if ( ! isset( $updates[ $wpjmcom_plugin_slug ] ) ) {
+				continue;
+			}
+
+			$response         = (object) $updates[ $wpjmcom_plugin_slug ];
+			$response->plugin = $filename;
+
 			// If there is a new version, modify the transient to reflect an update is available.
 			if (
-				$response
-				&& isset( $response['new_version'] )
-				&& ! empty( $response['new_version'] )
-				&& version_compare( $response['new_version'], $plugin_data['Version'], '>' )
+				! empty( $response->new_version )
+				&& version_compare( $response->new_version, $plugin_data['Version'], '>' )
 			) {
-				$available_addon_updates[ $product_slug ]                      = $response;
-				$check_for_updates_data->response[ $plugin_data['_filename'] ] = (object) $response;
+				$check_for_updates_data->response[ $plugin_data['_filename'] ] = $response;
+
+				$notice_data[] = [
+					'plugin'      => $response->plugin,
+					'plugin_name' => $response->plugin_name,
+					'new_version' => $response->new_version,
+				];
 			}
 		}
-		set_site_transient( 'wpjm_addon_updates_available', $available_addon_updates ); // No expiration set.
+
+		set_site_transient( 'wpjm_addon_updates_available', $notice_data );
 
 		return $check_for_updates_data;
 	}
 
 	/**
-	 * Get plugin version info from API.
+	 * Prepare the plugin update data for the WPJobManager.com request.
 	 *
-	 * @param string $plugin_filename
+	 * @param array $installed_plugins The array of installed WP Job Manager managed plugins.
 	 *
-	 * @return array|bool
+	 * @return array
+	 */
+	private function get_plugin_update_package( $installed_plugins ) {
+		$plugin_package = [];
+		foreach ( $installed_plugins as $plugin ) {
+			$plugin_slug = $plugin['_product_slug'];
+			$license_key = $this->get_plugin_licence( $plugin_slug );
+
+			$plugin_package[ $plugin_slug ] = [
+				'installed_version' => $plugin['Version'],
+				'license_key'       => $license_key['licence_key'] ?? '',
```

Comment:
Given that the default return value for `WP_Job_Manager_Helper_Options::get( $product_slug, 'licence_key' );` is `false`, I think that what we want here is: `$license_key['licence_key'] ? $license_key['licence_key'] : ''`

In my testings we were sending through the API: `..."wp-job-manager-embeddable-job-widget":{"installed_version":"1.1.3","license_key":false}...` which is returning a 400 because `license_key` is expected to be a string.

Reply:
Good catch! I changed downstream in the licence PR just so we can cleanly merge these. https://github.com/Automattic/WP-Job-Manager/pull/2554/commits/2f458247c094fcec74cbe203b596b29c535bb43a

That PR is up-to-date with this branch.